source/extensions/filters/http/composite/action.cc

```diff
@@ -15,34 +15,57 @@ Matcher::ActionFactoryCb ExecuteFilterActionFactory::createActionFactoryCb(
       const envoy::extensions::filters::http::composite::v3::ExecuteFilterAction&>(
       config, validation_visitor);
 
-  auto& factory =
-      Config::Utility::getAndCheckFactory<Server::Configuration::NamedHttpFilterConfigFactory>(
-          composite_action.typed_config());
-  ProtobufTypes::MessagePtr message = Config::Utility::translateAnyToFactoryConfig(
-      composite_action.typed_config().typed_config(), validation_visitor, factory);
-
-  Envoy::Http::FilterFactoryCb callback = nullptr;
-
-  // First, try to create the filter factory creation function from factory context (if exists).
-  if (context.factory_context_.has_value()) {
-    callback = factory.createFilterFactoryFromProto(*message, context.stat_prefix_,
-                                                    context.factory_context_.value());
+  if (composite_action.has_dynamic_config() && composite_action.has_typed_config()) {
+    throw EnvoyException(
+        fmt::format("Error: Only one of `dynamic_config` or `typed_config` can be set."));
   }
 
-  // If above failed, try to create the filter factory creation function from server factory
-  // context (if exists).
-  if (callback == nullptr && context.server_factory_context_.has_value()) {
-    callback = factory.createFilterFactoryFromProtoWithServerContext(
-        *message, context.stat_prefix_, context.server_factory_context_.value());
-  }
+  // Process dynamic filter configuration and setup extension configuration discovery service.
+  if (composite_action.has_dynamic_config()) {
+    auto config_discovery = composite_action.dynamic_config().config_discovery();
+    Server::Configuration::FactoryContext& factory_context = context.factory_context_.value();
+    // Create a dynamic filter config provider and register it with the server factory context.
+    auto provider = factory_context.createDynamicFilterConfigProvider(
+        config_discovery, composite_action.dynamic_config().name(), false, "http", nullptr);
+    return [provider = std::move(provider)]() -> Matcher::ActionPtr {
+      auto config_value = provider->config();
+      if (!config_value.has_value()) {
+        throw EnvoyException("Failed to get dynamic config for filter");
```

Comment:
Wouldn't this be called on datapath and crash?

Reply:
yeah. Was wondering how general ECDS handles this? what is your suggestion?

Reply:
It installs a dummy filter that always returns 5xx for HTTP.

Reply:
Ok. So should I remove this?

Reply:
I think it's OK to do the same thing as non-composite ECDS for HTTP, which is to return 500. The code is here:

https://github.com/envoyproxy/envoy/blob/05457bb047b87f4cc1a25635400ddcaa7fccc923/source/common/http/filter_chain_helper.cc#L54-L58

So instead of ExecuteFilterAction you need some action that always returns 500 I think.